views/charts.tmpl

```diff
@@ -90,61 +90,59 @@
         var currentGraph = "1";
 
         var ticketsFunc =function (){
-            tickets_price = "";
-            difficulty = "";
-            ticketsPrice.forEach(function(item){
-                tickets_price = tickets_price+ item.time+ ", "+ item.valuef+ "\n";
-                difficulty = difficulty+ item.time+ ", "+ item.sizef+ "\n";
+            tickets_price = [];
+            difficulty = [];
+            ticketsPrice.forEach(function(n){
+                tickets_price.push([new Date (n.time*1000), n.valuef]);
+                difficulty.push([new Date(n.time*1000), n.sizef]);
             });
         }
 
         var supplyFunc = function (){
             var supply = 0;
-            coin_supply = "";
-            coinSupply.forEach(function(item){
-                coin_supply = coin_supply + item.time+ ", "+ (isNaN(item.valuef) ? 0 : item.valuef) + "\n";
+            coin_supply = [];
+            coinSupply.forEach(function(n){
+                coin_supply.push([new Date(n.time*1000), (isNaN(n.valuef) ? 0 : n.valuef)]);
             });
         }
         
         var sizefunc = function (){
             var size = 0;
-            block_size = "";
-            blockchain_size = "";
-            tx_per_block = "";
-            time_btw_blocks = "";
-            blockSize.forEach(function(item){
-                block_size = block_size + item.time+ ", "+ item.size+ "\n";
-                size = size + item.size
-                blockchain_size = blockchain_size + item.time+ ", "+ size+"\n";
-                tx_per_block = tx_per_block + item.time + ", " + item.count+"\n";
-
-                if (isNaN(item.value)) {
-                    return;
-                }
-                time_btw_blocks = time_btw_blocks +  item.value + "," + (isNaN(item.valuef) ? 0 : item.valuef) + "\n";
+            block_size = [];
+            blockchain_size = [];
+            tx_per_block = [];
+            time_btw_blocks = [];
+            blockSize.forEach(function(n){
+                block_size.push([new Date(n.time*1000), n.size]);
```

Comment:
we can cache the result to avoid parsing same date more than once
eg `var date = new Date(n.time*1000)` then
```
block_size.push([date, n.size]);
blockchain_size.push([date, size]);
tx_per_block.push([date, n.count]);
```

Reply:
Noted.